src/hotspot/share/gc/shared/genArguments.cpp

```diff
@@ -304,9 +306,10 @@ void GenArguments::initialize_size_info() {
                             ", -XX:OldSize flag is being ignored",
                             MaxHeapSize);
       initial_old_size = MaxOldSize;
+    } else if (initial_old_size < MinOldSize) {
+      log_warning(gc, ergo)("Inconsistency between initial old size and minimum old size");
+      MinOldSize = initial_old_size;
```

Comment:
The comment shown below seems strange. Can it be removed?
```C++
    // The generation minimums and the overall heap minimum should
    // be within one generation alignment.
```


The warning message shown below can be more consistent with this newly added one. 
```C++
      log_warning(gc, ergo)("Inconsistency between maximum heap size and maximum "
                            "generation sizes: using maximum heap = " SIZE_FORMAT
                            ", -XX:OldSize flag is being ignored",
                            MaxHeapSize);
```

Reply:
If you think they are not related to this bug, it is also OK to fix them in another ticket.

Reply:
@lgxbslgx Thanks for the review, filed [JDK-8331432](https://bugs.openjdk.org/browse/JDK-8331432)